src/main/groovy/io/gatling/gradle/GatlingEnterpriseStartTask.groovy

```diff
@@ -9,45 +11,72 @@ import org.gradle.api.tasks.TaskExecutionException
 
 @CacheableTask
 class GatlingEnterpriseStartTask extends DefaultTask {
+    private static final String BATCH_MODE = "batchMode"
 
     @Internal
     Closure simulations
 
     @TaskAction
     void publish() {
         def gatling = project.extensions.getByType(GatlingPluginExtension)
-        EnterpriseClient enterpriseClient = gatling.enterprise.getEnterpriseClient(project.version.toString())
 
         def systemProps = gatling.enterprise.systemProps ?: [:]
 
-        if (!gatling.enterprise.simulationId && !gatling.enterprise.simulationClass) {
-            throw new TaskExecutionException(this,
-                new IllegalArgumentException("You need to either configure gatling.enterprise.simulationId if you want to start an existing simulation," +
-                "or gatling.enterprise.simulationClass if you want to create a new simulation")
-            )
-        }
-
         def simulationAndRunSummary
-        if (gatling.enterprise.simulationId) {
-            simulationAndRunSummary = enterpriseClient.startSimulation(gatling.enterprise.simulationId, systemProps, inputs.files.singleFile)
-        } else {
-            simulationAndRunSummary = enterpriseClient.createAndStartSimulation(
-                gatling.enterprise.teamId,
-                project.group.toString(),
-                project.name,
-                gatling.enterprise.simulationClass,
-                systemProps, inputs.files.singleFile)
-            def simulation = simulationAndRunSummary.simulation
-            getLogger().info("""
+
+        def batchMode = project.hasProperty(BATCH_MODE) ? Boolean.parseBoolean(project.getProperties().get(BATCH_MODE).toString()) : gatling.enterprise.batchMode
+
+        if (batchMode || gatling.enterprise.simulationId) {
+            if (!gatling.enterprise.simulationId && !gatling.enterprise.simulationClass) {
```

Comment:
`gatling.simulationClass` was decided
````suggestion
            if (!gatling.enterprise.simulationId && !gatling.simulationClass) {
```

Reply:
There is no param gatling.simulationClass for gatlingRun, so I set it directly into the enterprise config